bindings/java/iota-wallet-java/lib/src/main/java/org/iota/Wallet.java

```diff
@@ -286,4 +305,40 @@ public String hexToBech32(String hex, String bech32Hrp) throws WalletException {
         return callBaseApi(new WalletCommand("hexToBech32", p)).getAsString();
     }
 
+    /**
+     * Listen to wallet events, empty vec will listen to all events
+     * 
+     * @param listener The Listener object hat will handle events
+     * @param types    The types you want to listen. Empty means all events
+     * @throws WalletException
+     */
+    public void listen(EventListener listener, WalletEventType... types) throws WalletException {
+        callListen(listener, types);
+    }
+
+    /**
+     * Destroy the Wallet and drop its database connection.
+     * Unregisteres any existing listeners.
+     */
+    public void destroy() throws WalletException {
```

Comment:
This function destroys all Rust data but keeps the `Wallet` basically alive.
IMO this is not good, it shouldn't be possible to work with the `Wallet` after calling `destroy()`.
I guess it's not possible in Java to avoid that. 

What we could do:
Re-create automatically the Rust message handlers when working with the `Wallet` after calling `destroy()`. So basically on the Rust side, whenever a request comes in, check if the message handler is set. If not set create a new message handler. This would avoid user errors.

Reply:
yea this is not possible in java for sure. We could add a `boolean: destroyed` on `NativeAPI` and based on that re-call `createMessageHandler` but i think its weird if people manually call `destroy` and then continue using `Wallet`. Smells like they have some `async` methods running in the background. Sounds like bad logic on the user side to me.